art/attacks/poisoning/adversarial_embedding_attack.py

```diff
@@ -0,0 +1,296 @@
+        "discriminator_layer_2",
+        "regularization",
+        "learning_rate",
+    ]
+
+    # Currently only supporting Keras
+    _estimator_requirements = (BaseEstimator, NeuralNetworkMixin, ClassifierMixin)
+    ClassifierType = TypeVar('ClassifierType', bound=ClassifierNeuralNetwork)
+
+    def __init__(
+            self,
+            classifier: ClassifierType,
+            backdoor: PoisoningAttackBackdoor,
+            feature_layer: Union[int, str],
+            target: Union[np.ndarray, List[Tuple[np.ndarray, np.ndarray]]],
+            pp_poison: Union[float, List[float]] = 0.05,
+            discriminator_layer_1: int = 256,
+            discriminator_layer_2: int = 128,
+            regularization: float = 10,
+            learning_rate: float = 1e-4,
+    ):
+        """
+        Initialize an Feature Collision Clean-Label poisoning attack
+
+        :param classifier: A neural network classifier.
+        :param backdoor: The backdoor attack used to poison samples
+        :param feature_layer: The layer of the original network to extract features from
+        :param target: The target label to poison
+        :param pp_poison: The percentage of training data to poison
+        :param discriminator_layer_1: The size of the first discriminator layer
+        :param discriminator_layer_2: The size of the second discriminator layer
+        :param regularization: The regularization constant for the backdoor recognition part of the loss function
+        :param learning_rate: The learning rate of clean-label attack optimization.
+        """
+        super().__init__(classifier=classifier)
+        self.backdoor = backdoor
+        self.feature_layer = feature_layer
+        self.target = target
+        self.pp_poison = pp_poison
+        self.discriminator_layer_1 = discriminator_layer_1
+        self.discriminator_layer_2 = discriminator_layer_2
+        self.regularization = regularization
+        self.train_data: Optional[np.ndarray] = None
+        self.train_labels: Optional[np.ndarray] = None
+        self.is_backdoor: Optional[np.ndarray] = None
+        self.learning_rate = learning_rate
+        self._check_params()
+
+        if isinstance(self.estimator, KerasClassifier):
+            using_tf_keras = 'tensorflow.python.keras' in str(type(self.estimator.model))
+            if using_tf_keras:
+                from tensorflow.keras.models import Model, clone_model
+                from tensorflow.keras.layers import GaussianNoise, Dense, BatchNormalization, LeakyReLU
+                from tensorflow.keras.optimizers import Adam
+
+            else:
+                from keras import Model
+                from keras.models import clone_model
+                from keras.layers import GaussianNoise, Dense, BatchNormalization, LeakyReLU
+                from keras.optimizers import Adam
+
+            self.orig_model = clone_model(self.estimator.model, input_tensors=self.estimator.model.inputs)
```

Comment:
For models created with the functional API, the field input_tensors is ignored -- see https://github.com/tensorflow/tensorflow/issues/40617. Is it possible to make a copy without this method to support functional Keras models (e.g. [GTSRB micronnet](https://github.com/twosixlabs/armory/blob/master/armory/baseline_models/keras/micronnet_gtsrb.py)).

Reply:
Fixed this by adding a `clone=False` option for optionally allowing the changes to happen as a side effect until this is resolved upstream.